contracts/src/BeefyClient.sol

```diff
@@ -91,12 +97,14 @@ contract BeefyClient {
      * @param bitfield a bitfield signalling which validators they claim have signed
      * @param blockNumber the block number for this commitment
      * @param validatorSetLen the length of the validator set for this commitment
+     * @param numRequiredSignatures the number of signatures required
      * @param bitfield a bitfield signalling which validators they claim have signed
      */
     struct Ticket {
         address account;
         uint64 blockNumber;
         uint32 validatorSetLen;
+        uint256 numRequiredSignatures;
```

Comment:
As this ticket struct is persisted in storage, we should try and take care to pack the individual fields in as few bits as possible. This optimizes gas costs when writing and reading from storage: https://medium.com/coinmonks/gas-optimization-in-solidity-part-i-variables-9d5775e43dde

The `account`, `blockNumber`, and `validatorSetLen` fields all fit into a single 32-byte slot. 

However, we can optimize further by using these types:
```
uint16 validatorSetLen;
uint16 numRequiredSignatures;
``` 

Now the first four fields all fit in a 32-byte slot.

Since there are currently only 1000 validators on Kusama, I don't see the number of validators increasing by 65x without a major re-architecture of Polkadot consensus. And we can always deploy a new BeefyClient via governance if that happens.

Reply:
Actually it turns out that `ticket.account` isn't actually being used anywhere (unless I'm mistaken), so we can remove that field. Then we can leave `validatorSetLen` alone, and the only optimization we need to make is `uint32 numRequiredSignatures`.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/956/commits/5488978e8d91994bf7b027a5f82d4a41f30ff6c0.

Reply:
Hey Alistair, its dangerous to just cast the result of `computeNumSignatures` to uint32, as it could result in undefined output.

While I know that practically the numSignatures value will never ever require more than 32 bits, auditors are still going to point this out.

Instead, I would add a `Math.min(type(uint32).max, ...)` as the last expression in `computeNumSignatures`, to clamp the maximum value to `type(uint32).max`.

Then the cast is safe.